Code/GraphMol/MolProcessing/MolProcessing.cpp

```diff
@@ -0,0 +1,156 @@
+//
+//  Copyright (C) 2024 Greg Landrum and other RDKit contributors
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#include "Processing.h"
+
+namespace RDKit {
+namespace MolProcessing {
+
+namespace details {
+GeneralMolSupplier::SupplierOptions defaultSupplierOptions;
+inline std::unique_ptr<FileParsers::MolSupplier> getSupplier(
+    const std::string &fileName,
+    const GeneralMolSupplier::SupplierOptions &options) {
+#ifdef RDK_BUILD_THREADSAFE_SSS
+  static std::once_flag flag;
+  std::call_once(flag, []() { defaultSupplierOptions.numWriterThreads = 4; });
+#endif
+  return GeneralMolSupplier::getSupplier(fileName, options);
+}
+}  // namespace details
+
+namespace {
+#ifdef RDK_BUILD_THREADSAFE_SSS
+inline std::mutex &fp_mutex_get() {
+  // create on demand
+  static std::mutex _mutex;
+  return _mutex;
+}
+
+inline void fp_mutex_create() {
+  std::mutex &mutex = fp_mutex_get();
+  std::lock_guard<std::mutex> test_lock(mutex);
+}
+
+inline std::mutex &get_fp_mutex() {
+  static std::once_flag flag;
```

Comment:
I'm not sure this is necessary, in c++11 and above, 

```
inline std::mutex fp_mutex_get() {
static std::mutex _mutex;
...
}
```

is created in a thread safe manner (It basically is required to have call_once semantics).
 
I might be confused about the semantics of the calls here though.

Reply:
Nope, you're right. There's a lot of un-necessary code here. I will fix it.